### ISSUE TITLE
rvm install 2.1.1 - failing on cygwin

### ISSUE THREAD

**Issue by marcelomazza** [rvm/rvm#3358]
Hello, hope I'm posting correctly here :) I'm trying to install ruby 2.1.1 with cygwin and tried a lot of things and I couldn't make it work.

Initially, I followed this guide http://blog.developwithpassion.com/2012/03/30/installing-rvm-with-cygwin-on-windows/

I tried with `stable` and `head`, also these is a list of commands I tried:

```
rvm reinstall 2.1.1 --disable-binary

rvm autolibs disable

rvm get branch /features/cygwin
rvm install 2.1.0

rvm install 2.1.0 --patch https://github.com/kou1okada/ruby/commit/b05b4f1e0f45c00a235e487fb20e7a6a9e74a43b?format=diff

rvm install 2.1.1 -C --with-readline-dir=$HOME/.rvm/usr
```

None of them worked :( and all of them give me the same error (see below).

**Thank you very much for your help!**

Here is the output in a gist, and also the code block:
https://gist.github.com/marcelomazza/b741237132c60a615e7f
```
$ rvm --debug install 2.1.1
Warning: No 'sudo' found.
Warning: No 'sudo' found.
ruby-2.1.1 - install
ruby-2.1.1 - #already removed src/ruby-2.1.1
ruby-2.1.1 - #already removed rubies/ruby-2.1.1
Searching for binary rubies, this might take some time.
Remote file does not exist https://rvm_io.global.ssl.fastly.net/binaries/cygwin/unknown/i386/ruby-2.1.1.tar.bz2
Remote file does not exist https://s3.amazonaws.com/jruby.org/downloads/ruby-2.1.1.tar.bz2
Remote file does not exist http://binaries.rubini.us/cygwin/unknown/i386/ruby-2.1.1.tar.bz2
Remote file does not exist https://rubies.travis-ci.org/cygwin/unknown/i386/ruby-2.1.1.tar.bz2
rvm_remote_server_url4 not found
No remote file name found
Remote file does not exist https://rvm_io.global.ssl.fastly.net/binaries/cygwin/unknown/i386/ruby-2.1.1.tar.bz2
Remote file does not exist https://s3.amazonaws.com/jruby.org/downloads/ruby-2.1.1.tar.bz2
Remote file does not exist http://binaries.rubini.us/cygwin/unknown/i386/ruby-2.1.1.tar.bz2
Remote file does not exist https://rubies.travis-ci.org/cygwin/unknown/i386/ruby-2.1.1.tar.bz2
rvm_remote_server_url4 not found
No remote file name found
No binary rubies available for: cygwin/unknown/i386/ruby-2.1.1.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Free disk space 180044MB, required 440MB.
__rvm_setup_compile_environment_setup ruby-2.1.1
rvm_autolibs_flag=disable
__rvm_setup_compile_environment_movable_early ruby-2.1.1
__rvm_setup_compile_environment_system_early ruby-2.1.1
__rvm_setup_compile_environment_requirements ruby-2.1.1
__rvm_setup_compile_environment_movable ruby-2.1.1
__rvm_setup_compile_environment_parse_name ruby-2.1.1
__rvm_setup_compile_environment_system ruby-2.1.1
__rvm_setup_compile_environment_architectures ruby-2.1.1
rvm_architectures(0):.
__rvm_setup_compile_environment_gcc47plus ruby-2.1.1
__rvm_setup_compile_environment_bison ruby-2.1.1
__rvm_setup_compile_environment_flags_docs ruby-2.1.1
__rvm_setup_compile_environment_flags_shared_static ruby-2.1.1
__rvm_setup_compile_environment_flags_threads ruby-2.1.1
__rvm_setup_compile_environment_compatibility_flag ruby-2.1.1
found compiler: /usr/bin/gcc
Installing Ruby from source to: /home/mnm/.rvm/rubies/ruby-2.1.1, this may take a while depending on your cpu(s)...
ruby-2.1.1 - #downloading ruby-2.1.1, this may take a while depending on your connection...
Warning: No 'sudo' found.
fetch http://cache.ruby-lang.org/pub/ruby/2.1/ruby-2.1.1.tar.bz2 => ruby-2.1.1.tar.bz2
Archive checksum matched, not downloading
Downloaded archive checksum matched.
Log prefix: /home/mnm/.rvm/log/1427472143_ruby-2.1.1/
Log filesystem: ntfs
Log max name length: 250
Log file: /home/mnm/.rvm/log/1427472143_ruby-2.1.1/extract.log
[2015-03-27 17:02:24] __rvm_package_extract
__rvm_package_extract ()
{
    rvm_debug __rvm_package_extract:$#: "$@";
    \typeset __tempdir __path __file __return;
    __return=0;
    __tempdir="$( TMPDIR="${rvm_tmp_path}" mktemp -d -t rvm-tmp.XXXXXXXXX )";
    __rvm_package_extract_run "$1" "$__tempdir" || __return=$?;
    if (( __return == 0 )); then
        for __path in "$__tempdir"/*;
        do
            __file="${__path##*/}";
            if [[ -n "${__file}" && -e "$2/${__file}" ]]; then
                \command \rm -rf "$2/${__file}" || __return=$?;
            fi;
            \command \mv -f "${__path}" "$2/" || __return=$?;
        done;
    fi;
    if [[ -n "$__tempdir" ]]; then
        \command \rm -rf "$__tempdir";
    fi;
    return $__return
}
current path: /home/mnm/.rvm/src
PATH=/usr/local/bin:/usr/bin:/cygdrive/c/Program Files (x86)/NVIDIA Corporation/PhysX/Common:/cygdrive/c/Windows/system32:/cygdrive/c/Windows:/cygdrive/c/Windows/System32/Wbem:/cygdrive/c/Windows/System32/WindowsPowerShell/v1.0:/cygdrive/c/Program Files (x86)/Intel/OpenCL SDK/2.0/bin/x86:/cygdrive/c/Program Files (x86)/Intel/OpenCL SDK/2.0/bin/x64:/cygdrive/c/Program Files (x86)/Skype/Phone:/cygdrive/c/Program Files (x86)/gdipp:/cygdrive/c/Program Files/nodejs:/cygdrive/c/Program Files (x86)/NVIDIA Corporation/PhysX/Common:/cygdrive/c/Windows/system32:/cygdrive/c/Windows:/cygdrive/c/Windows/System32/Wbem:/cygdrive/c/Windows/System32/WindowsPowerShell/v1.0:/cygdrive/c/Program Files (x86)/Intel/OpenCL SDK/2.0/bin/x86:/cygdrive/c/Program Files (x86)/Intel/OpenCL SDK/2.0/bin/x64:/cygdrive/c/Program Files (x86)/Skype/Phone:/cygdrive/c/sites/uru:/cygdrive/c/Users/mnm/AppData/Roaming/npm:/usr/lib/lapack:/home/mnm/.rvm/bin
command(3): __rvm_package_extract /home/mnm/.rvm/archives/ruby-2.1.1.tar.bz2 /home/mnm/.rvm/tmp/rvm_src_2384
ruby-2.1.1 - #extracting ruby-2.1.1 to /home/mnm/.rvm/src/ruby-2.1.1.....
__rvm_rm_rf already gone: /home/mnm/.rvm/src/ruby-2.1.1
Patch name 'optional', expanded_name(s) ''.
Patch name 'default', expanded_name(s) 'libyaml changeset_r45225 changeset_r45240'.
Patch name 'cygwin', expanded_name(s) ''.
All found patches(3): libyaml changeset_r45225 changeset_r45240.
Patch full path '/home/mnm/.rvm/patches/ruby/2.1.1/libyaml.patch'.
Log file: /home/mnm/.rvm/log/1427472143_ruby-2.1.1/patch.apply.libyaml.log
[2015-03-27 17:03:15] __rvm_patch
__rvm_patch ()
{
    \patch "$@" || return $?
}
current path: /home/mnm/.rvm/src/ruby-2.1.1
PATH=/home/mnm/.rvm/usr/bin:/usr/local/bin:/usr/bin:/cygdrive/c/Program Files (x86)/NVIDIA Corporation/PhysX/Common:/cygdrive/c/Windows/system32:/cygdrive/c/Windows:/cygdrive/c/Windows/System32/Wbem:/cygdrive/c/Windows/System32/WindowsPowerShell/v1.0:/cygdrive/c/Program Files (x86)/Intel/OpenCL SDK/2.0/bin/x86:/cygdrive/c/Program Files (x86)/Intel/OpenCL SDK/2.0/bin/x64:/cygdrive/c/Program Files (x86)/Skype/Phone:/cygdrive/c/Program Files (x86)/gdipp:/cygdrive/c/Program Files/nodejs:/cygdrive/c/Program Files (x86)/NVIDIA Corporation/PhysX/Common:/cygdrive/c/Windows/system32:/cygdrive/c/Windows:/cygdrive/c/Windows/System32/Wbem:/cygdrive/c/Windows/System32/WindowsPowerShell/v1.0:/cygdrive/c/Program Files (x86)/Intel/OpenCL SDK/2.0/bin/x86:/cygdrive/c/Program Files (x86)/Intel/OpenCL SDK/2.0/bin/x64:/cygdrive/c/Program Files (x86)/Skype/Phone:/cygdrive/c/sites/uru:/cygdrive/c/Users/mnm/AppData/Roaming/npm:/usr/lib/lapack:/home/mnm/.rvm/bin
command(8): __rvm_patch -F 25 -p1 -N -f -i /home/mnm/.rvm/patches/ruby/2.1.1/libyaml.patch
ruby-2.1.1 - #applying patch /home/mnm/.rvm/patches/ruby/2.1.1/libyaml.patch.
Patch full path '/home/mnm/.rvm/patches/ruby/changeset_r45225.diff'.
Log file: /home/mnm/.rvm/log/1427472143_ruby-2.1.1/patch.apply.changeset_r45225.log
[2015-03-27 17:03:16] __rvm_patch
__rvm_patch ()
{
    \patch "$@" || return $?
}
current path: /home/mnm/.rvm/src/ruby-2.1.1
PATH=/home/mnm/.rvm/usr/bin:/usr/local/bin:/usr/bin:/cygdrive/c/Program Files (x86)/NVIDIA Corporation/PhysX/Common:/cygdrive/c/Windows/system32:/cygdrive/c/Windows:/cygdrive/c/Windows/System32/Wbem:/cygdrive/c/Windows/System32/WindowsPowerShell/v1.0:/cygdrive/c/Program Files (x86)/Intel/OpenCL SDK/2.0/bin/x86:/cygdrive/c/Program Files (x86)/Intel/OpenCL SDK/2.0/bin/x64:/cygdrive/c/Program Files (x86)/Skype/Phone:/cygdrive/c/Program Files (x86)/gdipp:/cygdrive/c/Program Files/nodejs:/cygdrive/c/Program Files (x86)/NVIDIA Corporation/PhysX/Common:/cygdrive/c/Windows/system32:/cygdrive/c/Windows:/cygdrive/c/Windows/System32/Wbem:/cygdrive/c/Windows/System32/WindowsPowerShell/v1.0:/cygdrive/c/Program Files (x86)/Intel/OpenCL SDK/2.0/bin/x86:/cygdrive/c/Program Files (x86)/Intel/OpenCL SDK/2.0/bin/x64:/cygdrive/c/Program Files (x86)/Skype/Phone:/cygdrive/c/sites/uru:/cygdrive/c/Users/mnm/AppData/Roaming/npm:/usr/lib/lapack:/home/mnm/.rvm/bin
command(8): __rvm_patch -F 25 -p1 -N -f -i /home/mnm/.rvm/patches/ruby/changeset_r45225.diff
ruby-2.1.1 - #applying patch /home/mnm/.rvm/patches/ruby/changeset_r45225.diff.
Patch full path '/home/mnm/.rvm/patches/ruby/changeset_r45240.diff'.
Log file: /home/mnm/.rvm/log/1427472143_ruby-2.1.1/patch.apply.changeset_r45240.log
[2015-03-27 17:03:16] __rvm_patch
__rvm_patch ()
{
    \patch "$@" || return $?
}
current path: /home/mnm/.rvm/src/ruby-2.1.1
PATH=/home/mnm/.rvm/usr/bin:/usr/local/bin:/usr/bin:/cygdrive/c/Program Files (x86)/NVIDIA Corporation/PhysX/Common:/cygdrive/c/Windows/system32:/cygdrive/c/Windows:/cygdrive/c/Windows/System32/Wbem:/cygdrive/c/Windows/System32/WindowsPowerShell/v1.0:/cygdrive/c/Program Files (x86)/Intel/OpenCL SDK/2.0/bin/x86:/cygdrive/c/Program Files (x86)/Intel/OpenCL SDK/2.0/bin/x64:/cygdrive/c/Program Files (x86)/Skype/Phone:/cygdrive/c/Program Files (x86)/gdipp:/cygdrive/c/Program Files/nodejs:/cygdrive/c/Program Files (x86)/NVIDIA Corporation/PhysX/Common:/cygdrive/c/Windows/system32:/cygdrive/c/Windows:/cygdrive/c/Windows/System32/Wbem:/cygdrive/c/Windows/System32/WindowsPowerShell/v1.0:/cygdrive/c/Program Files (x86)/Intel/OpenCL SDK/2.0/bin/x86:/cygdrive/c/Program Files (x86)/Intel/OpenCL SDK/2.0/bin/x64:/cygdrive/c/Program Files (x86)/Skype/Phone:/cygdrive/c/sites/uru:/cygdrive/c/Users/mnm/AppData/Roaming/npm:/usr/lib/lapack:/home/mnm/.rvm/bin
command(8): __rvm_patch -F 25 -p1 -N -f -i /home/mnm/.rvm/patches/ruby/changeset_r45240.diff
ruby-2.1.1 - #applying patch /home/mnm/.rvm/patches/ruby/changeset_r45240.diff.
Log file: /home/mnm/.rvm/log/1427472143_ruby-2.1.1/configure.log
[2015-03-27 17:03:16] ./configure
current path: /home/mnm/.rvm/src/ruby-2.1.1
PATH=/home/mnm/.rvm/usr/bin:/usr/local/bin:/usr/bin:/cygdrive/c/Program Files (x86)/NVIDIA Corporation/PhysX/Common:/cygdrive/c/Windows/system32:/cygdrive/c/Windows:/cygdrive/c/Windows/System32/Wbem:/cygdrive/c/Windows/System32/WindowsPowerShell/v1.0:/cygdrive/c/Program Files (x86)/Intel/OpenCL SDK/2.0/bin/x86:/cygdrive/c/Program Files (x86)/Intel/OpenCL SDK/2.0/bin/x64:/cygdrive/c/Program Files (x86)/Skype/Phone:/cygdrive/c/Program Files (x86)/gdipp:/cygdrive/c/Program Files/nodejs:/cygdrive/c/Program Files (x86)/NVIDIA Corporation/PhysX/Common:/cygdrive/c/Windows/system32:/cygdrive/c/Windows:/cygdrive/c/Windows/System32/Wbem:/cygdrive/c/Windows/System32/WindowsPowerShell/v1.0:/cygdrive/c/Program Files (x86)/Intel/OpenCL SDK/2.0/bin/x86:/cygdrive/c/Program Files (x86)/Intel/OpenCL SDK/2.0/bin/x64:/cygdrive/c/Program Files (x86)/Skype/Phone:/cygdrive/c/sites/uru:/cygdrive/c/Users/mnm/AppData/Roaming/npm:/usr/lib/lapack:/home/mnm/.rvm/bin
command(4): ./configure --prefix=/home/mnm/.rvm/rubies/ruby-2.1.1 --disable-install-doc --enable-shared
ruby-2.1.1 - #configuring...................................................
Log file: /home/mnm/.rvm/log/1427472143_ruby-2.1.1/postconfigure.log
[2015-03-27 17:06:04] __rvm_post_configure_ruby
__rvm_post_configure_ruby ()
{
    __rvm_post_configure_ruby_fix_athlon_sse2;
    __rvm_post_configure_ruby_update_setup
}
current path: /home/mnm/.rvm/src/ruby-2.1.1
PATH=/home/mnm/.rvm/usr/bin:/usr/local/bin:/usr/bin:/cygdrive/c/Program Files (x86)/NVIDIA Corporation/PhysX/Common:/cygdrive/c/Windows/system32:/cygdrive/c/Windows:/cygdrive/c/Windows/System32/Wbem:/cygdrive/c/Windows/System32/WindowsPowerShell/v1.0:/cygdrive/c/Program Files (x86)/Intel/OpenCL SDK/2.0/bin/x86:/cygdrive/c/Program Files (x86)/Intel/OpenCL SDK/2.0/bin/x64:/cygdrive/c/Program Files (x86)/Skype/Phone:/cygdrive/c/Program Files (x86)/gdipp:/cygdrive/c/Program Files/nodejs:/cygdrive/c/Program Files (x86)/NVIDIA Corporation/PhysX/Common:/cygdrive/c/Windows/system32:/cygdrive/c/Windows:/cygdrive/c/Windows/System32/Wbem:/cygdrive/c/Windows/System32/WindowsPowerShell/v1.0:/cygdrive/c/Program Files (x86)/Intel/OpenCL SDK/2.0/bin/x86:/cygdrive/c/Program Files (x86)/Intel/OpenCL SDK/2.0/bin/x64:/cygdrive/c/Program Files (x86)/Skype/Phone:/cygdrive/c/sites/uru:/cygdrive/c/Users/mnm/AppData/Roaming/npm:/usr/lib/lapack:/home/mnm/.rvm/bin
command(1): __rvm_post_configure_ruby
ruby-2.1.1 - #post-configuration..
Log file: /home/mnm/.rvm/log/1427472143_ruby-2.1.1/make.log
[2015-03-27 17:06:04] __rvm_make
__rvm_make ()
{
    \make "$@" || return $?
}
current path: /home/mnm/.rvm/src/ruby-2.1.1
PATH=/home/mnm/.rvm/usr/bin:/usr/local/bin:/usr/bin:/cygdrive/c/Program Files (x86)/NVIDIA Corporation/PhysX/Common:/cygdrive/c/Windows/system32:/cygdrive/c/Windows:/cygdrive/c/Windows/System32/Wbem:/cygdrive/c/Windows/System32/WindowsPowerShell/v1.0:/cygdrive/c/Program Files (x86)/Intel/OpenCL SDK/2.0/bin/x86:/cygdrive/c/Program Files (x86)/Intel/OpenCL SDK/2.0/bin/x64:/cygdrive/c/Program Files (x86)/Skype/Phone:/cygdrive/c/Program Files (x86)/gdipp:/cygdrive/c/Program Files/nodejs:/cygdrive/c/Program Files (x86)/NVIDIA Corporation/PhysX/Common:/cygdrive/c/Windows/system32:/cygdrive/c/Windows:/cygdrive/c/Windows/System32/Wbem:/cygdrive/c/Windows/System32/WindowsPowerShell/v1.0:/cygdrive/c/Program Files (x86)/Intel/OpenCL SDK/2.0/bin/x86:/cygdrive/c/Program Files (x86)/Intel/OpenCL SDK/2.0/bin/x64:/cygdrive/c/Program Files (x86)/Skype/Phone:/cygdrive/c/sites/uru:/cygdrive/c/Users/mnm/AppData/Roaming/npm:/usr/lib/lapack:/home/mnm/.rvm/bin
command(2): __rvm_make -j8
ruby-2.1.1 - #compiling....................................................
Error running '__rvm_make -j8',
showing last 15 lines of /home/mnm/.rvm/log/1427472143_ruby-2.1.1/make.log
installing default path_to_class libraries
installing default rational libraries
linking shared-object -test-/path_to_class/path_to_class.so
linking shared-object -test-/postponed_job.so
make[2]: Leaving directory '/home/mnm/.rvm/src/ruby-2.1.1/ext/-test-/num2int'
make[2]: Leaving directory '/home/mnm/.rvm/src/ruby-2.1.1/ext/-test-/old_thread_select'
linking shared-object -test-/rational.so
make[2]: Leaving directory '/home/mnm/.rvm/src/ruby-2.1.1/ext/-test-/method'
make[2]: Leaving directory '/home/mnm/.rvm/src/ruby-2.1.1/ext/-test-/path_to_class'
make[2]: Leaving directory '/home/mnm/.rvm/src/ruby-2.1.1/ext/-test-/postponed_job'
make[2]: Leaving directory '/home/mnm/.rvm/src/ruby-2.1.1/ext/-test-/rational'
make[1]: Leaving directory '/home/mnm/.rvm/src/ruby-2.1.1'
uncommon.mk:180: recipe for target 'build-ext' failed
make: *** [build-ext] Error 2
++ return 2
There has been an error while running make. Halting the installation.
__rvm_rm_rf already gone: /home/mnm/.rvm/tmp/2384*
```

**Comment by samneely** [comment]
According to the FAQs on RVM.io, RVM does not currently support the Windows operating system; however, there are plans to support Windows with RVM 2.0. http://rvm.io/support/faq#does-rvm-work-on-windows-will-it-in-the-future

**Comment by havenwood** [comment]
@samneely Though RVM1 doesn't have native Windows support it's known to work with Cygwin. (Although apparently not in this case!)

@marcelomazza I wonder if this is a dup of #3309 or a separate issue? Could you show the end of the `rvm --trace install ruby-2.1.1` output?

**Comment by havenwood** [comment]
@marcelomazza Does `rvm install 2.1.5` succeed?

**Comment by marcelomazza** [comment]
Thanks guys for the help!

@havenwood this is the output of `rvm --trace install ruby 2.1.1`: https://gist.github.com/marcelomazza/be093fe9f9946a02b6f6

I'll try with 2.1.5 and post the results.

**Comment by marcelomazza** [comment]
@havenwood  `rvm install 2.1.5` also gives me an error :(

https://gist.github.com/marcelomazza/b857a07a9a296d880ba1

**Comment by d4rky-pl** [comment]
@marcelomazza it seems that it's a bug in Ruby, not in RVM - https://bugs.ruby-lang.org/issues/11065

**Comment by CMCDragonkai** [comment]
I'm getting a similar error:

```
Searching for binary rubies, this might take some time.
No binary rubies available for: cygwin/unknown/x86_64/ruby-2.2.1.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for cygwin.
Installing requirements for cygwin.
Updating system.
Installing required packages: mingw64-i686-gcc, mingw64-x86_64-gcc.
Requirements installation successful.
Found user configured '-j' flag in 'rvm_make_flags', please note that RVM can detect number of CPU threads and set the '-j' flag automatically if you do not set it.
Installing Ruby from source to: /cygdrive/c/Users/CMCDragonkai/.rvm/rubies/ruby-2.2.1, this may take a while depending on your cpu(s)...
ruby-2.2.1 - #downloading ruby-2.2.1, this may take a while depending on your connection...
ruby-2.2.1 - #extracting ruby-2.2.1 to /cygdrive/c/Users/CMCDragonkai/.rvm/src/ruby-2.2.1....
ruby-2.2.1 - #applying patch /cygdrive/c/Users/CMCDragonkai/.rvm/patches/ruby/2.2.1/fix_installing_bundled_gems.patch.
ruby-2.2.1 - #configuring.......................................................
ruby-2.2.1 - #post-configuration..
ruby-2.2.1 - #compiling...............................................
Error running '__rvm_make -j2',
showing last 15 lines of /cygdrive/c/Users/CMCDragonkai/.rvm/log/1430397941_ruby-2.2.1/make.log
make[2]: *** [printf.o] Error 1
make[2]: Leaving directory '/cygdrive/c/Users/CMCDragonkai/.rvm/src/ruby-2.2.1/ext/-test-/printf'
exts.mk:118: recipe for target 'ext/-test-/printf/all' failed
make[1]: *** [ext/-test-/printf/all] Error 2
make[1]: *** Waiting for unfinished jobs....
make[2]: Entering directory '/cygdrive/c/Users/CMCDragonkai/.rvm/src/ruby-2.2.1/ext/-test-/proc'
compiling init.c
compiling receiver.c
compiling super.c
linking shared-object -test-/proc.so
make[2]: Leaving directory '/cygdrive/c/Users/CMCDragonkai/.rvm/src/ruby-2.2.1/ext/-test-/proc'
make[1]: Leaving directory '/cygdrive/c/Users/CMCDragonkai/.rvm/src/ruby-2.2.1'
uncommon.mk:189: recipe for target 'build-ext' failed
make: *** [build-ext] Error 2
++ /scripts/functions/support : __rvm_make()  383 > return 2
There has been an error while running make. Halting the installation.
```

@d4rky-pl Has that fix arrived in RVM?

**Comment by d4rky-pl** [comment]
@CMCDragonkai like I said it seems to be a problem with Ruby rather than RVM. You can try making a patch from https://bugs.ruby-lang.org/projects/ruby-trunk/repository/revisions/50305 and applying it to Ruby source code before compilation. 

I currently don't have any Windows machine to test it on Cygwin so you're on your own, at least for now (or until Ruby backports this fix).

**Comment by havenwood** [comment]
Closing as resolved upstream with the release of 2.1.7 and 2.2.3.
